src/libcmd/installables.cc

```diff
@@ -615,42 +618,118 @@ std::tuple<std::string, FlakeRef, InstallableValue::DerivationInfo> InstallableF
+        }
 
-    if (outputsToInstall.empty() || std::get_if<AllOutputs>(&outputsSpec)) {
-        outputsToInstall.clear();
-        if (auto aOutputs = attr->maybeGetAttr(state->sOutputs))
-            for (auto & s : aOutputs->getListOfStrings())
-                outputsToInstall.insert(s);
+        if (outputsToInstall.empty() || std::get_if<AllOutputs>(&outputsSpec)) {
+            outputsToInstall.clear();
+            if (auto aOutputs = attr->maybeGetAttr(state->sOutputs))
+                for (auto & s : aOutputs->getListOfStrings())
+                    outputsToInstall.insert(s);
+        }
+
+        if (outputsToInstall.empty())
+            outputsToInstall.insert("out");
+
+        if (auto outputNames = std::get_if<OutputNames>(&outputsSpec))
+            outputsToInstall = *outputNames;
+
+        auto drvInfo = DerivationInfo {
+            .drvPath = std::move(drvPath),
+            .outputsToInstall = std::move(outputsToInstall),
+            .priority = priority,
+        };
+
+        return {attrPath, getLockedFlake()->flake.lockedRef, std::move(drvInfo)};
     }
 
-    if (outputsToInstall.empty())
-        outputsToInstall.insert("out");
+    else if (type == "configurable") {
 
-    if (auto outputNames = std::get_if<OutputNames>(&outputsSpec))
-        outputsToInstall = *outputNames;
+        // FIXME: use eval cache
 
-    auto drvInfo = DerivationInfo {
-        .drvPath = std::move(drvPath),
-        .outputsToInstall = std::move(outputsToInstall),
-        .priority = priority,
-    };
+        auto & build = attr->getAttr("build")->forceValue();
 
-    return {attrPath, getLockedFlake()->flake.lockedRef, std::move(drvInfo)};
+        struct ValueTree
+        {
+            std::map<std::string, std::variant<Value *, ValueTree>> children;
+        };
+
+        ValueTree tree;
+
+        if (cmd) {
+            auto autoArgs = cmd->getAutoArgs(*state);
+
+            for (auto & attr : *autoArgs) {
+                auto ss = tokenizeString<std::vector<std::string>>(state->symbols[attr.name], ".");
+                auto * pos = &tree;
+                for (const auto & [n, s] : enumerate(ss)) {
+                    if (n + 1 == ss.size()) {
+                        if (pos->children.find(s) != pos->children.end())
+                            throw Error("definition of '%s' clashes with a previous definition", state->symbols[attr.name]);
+                        pos->children.emplace(s, attr.value);
+                    } else {
+                        auto i = pos->children.emplace(s, ValueTree()).first;
+                        if (auto pos2 = std::get_if<ValueTree>(&i->second))
+                            pos = pos2;
+                        else
+                            throw Error("definition of '%s' clashes with a previous definition", state->symbols[attr.name]);
+                    }
+                }
+            }
+        }
+
+        std::function<Value * (const ValueTree & tree)> buildAttrs;
```

Comment:
Might be worth a comment that this declared here for recursive calling, it surprised me.